src/Microsoft.AspNetCore.Routing/DefaultLinkGenerator.cs

```diff
@@ -234,7 +247,41 @@ private RouteValueDictionary GetAmbientValues(HttpContext httpContext)
                     return feature.Values;
                 }
             }
-            return null;
+            return new RouteValueDictionary();
+        }
+
+        public override LinkGenerationTemplate GetTemplate(HttpContext httpContext, object values, LinkOptions options)
```

Comment:
This should probably start with a `TAddress` version, and then specialize from there